hack/install_kustomize.sh

```diff
@@ -95,13 +95,17 @@ trap cleanup EXIT ERR
 pushd "$tmpDir" >& /dev/null
 
 opsys=windows
-arch=amd64
 if [[ "$OSTYPE" == linux* ]]; then
   opsys=linux
 elif [[ "$OSTYPE" == darwin* ]]; then
   opsys=darwin
 fi
 
+arch=amd64
+if command -v arch &> /dev/null; then
```

Comment:
I looked into whether this is a POSIX standard tool, and unfortunately it isn't. It seems like the correct thing to use is `uname -m`, [which is standard](https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/utilities/uname.html). However, there's a further complication: the values it returns do not match up with the names our release pipeline uses on the artifacts. I only have one data point in support of this, but it is an important one: on an Intel Macbook, `uname -m` returns `x86_64` (and `arch` returns `i386` fwiw). 

I suggest we take advantage of the fact that there are a very small number of values that will lead to a valid download path here. Start with `uname -m`, map the output to the equivalent goreleaser values when the conversion is well known, and then have a final check against the list of supported names, which falls back on the current default in case of no match. Wdyt?

Reply:
Makes sense to me.  I'll post a follow up we can try with this approach.

Reply:
I'm posted an update, I haven't been able to test this yet on other systems yet, so at moment only tried this on Apple M1 so arm64.   I'll try this on an Intel Mac and Intel Linux later tonight.  I need to confirm the output of `uname -m` on ppc64e and s390x, I just guessed for now.  Will mark this PR as WIP until I can test some more.

Reply:
SG, thank you. Ping me when you're ready for another review.